map/framework.cpp

```diff
@@ -2472,7 +2477,9 @@ void Framework::SetRouterImpl(RouterType type)
 
     router.reset(
         new CarRouter(m_model.GetIndex(), countryFileGetter,
-                      SingleMwmRouter::CreateCarRouter(m_model.GetIndex())));
+                      SingleMwmRouter::CreateCarRouter(m_model.GetIndex(),
+                                                       bind(&RoutingSession::GetTrafficColoring,
```

Comment:
So, it seems that you pass anonymous object here, and store const reference to it. Looks like a direct way to segfault.